core/src/main/java/org/infinispan/persistence/manager/TracedPersistenceManager.java

```diff
@@ -0,0 +1,117 @@
+                                                       Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("deleteFromAllStores", super.deleteFromAllStores(key, segment, predicate));
+   }
+
+   @Override
+   public <K, V> CompletionStage<MarshallableEntry<K, V>> loadFromAllStores(Object key, int segment, boolean localInvocation, boolean includeStores) {
+      return decorateCommand("loadFromAllStores", super.loadFromAllStores(key, segment, localInvocation, includeStores));
+   }
+
+   @Override
+   public CompletionStage<Long> approximateSize(Predicate<? super StoreConfiguration> predicate, IntSet segments) {
+      return decorateCommand("approximateSize", super.approximateSize(predicate, segments));
+   }
+
+   @Override
+   public CompletionStage<Long> size(Predicate<? super StoreConfiguration> predicate, IntSet segments) {
+      return decorateCommand("size", super.size(predicate, segments));
+   }
+
+   @Override
+   public CompletionStage<Void> writeToAllNonTxStores(MarshallableEntry marshalledEntry, int segment,
+                                                      Predicate<? super StoreConfiguration> predicate, long flags) {
+      return decorateCommand("writeToAllNonTxStores", super.writeToAllNonTxStores(marshalledEntry, segment, predicate, flags));
+   }
+
+   @Override
+   public CompletionStage<Void> commitAllTxStores(TxInvocationContext<AbstractCacheTransaction> txInvocationContext,
+                                                  Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("commitAllTxStores", super.commitAllTxStores(txInvocationContext, predicate));
+   }
+
+   @Override
+   public CompletionStage<Void> rollbackAllTxStores(TxInvocationContext<AbstractCacheTransaction> txInvocationContext,
+                                                    Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("rollbackAllTxStores", super.rollbackAllTxStores(txInvocationContext, predicate));
+   }
+
+   @Override
+   public <K, V> CompletionStage<Void> writeEntries(Iterable<MarshallableEntry<K, V>> iterable,
+                                                    Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("writeEntries", super.writeEntries(iterable, predicate));
+   }
+
+   @Override
+   public CompletionStage<Long> performBatch(TxInvocationContext<AbstractCacheTransaction> ctx,
+                                             TriPredicate<? super WriteCommand, Object, MVCCEntry<?, ?>> commandKeyPredicate) {
+      return decorateCommand("performBatch", super.performBatch(ctx, commandKeyPredicate));
+   }
+
+   private InfinispanSpan requestStart(String operationName) {
+      InfinispanSpanAttributes attributes = new InfinispanSpanAttributes.Builder(SpanCategory.PERSISTENCE)
+            .withCache(cache.getName(), configuration)
+            .build();
```

Comment:
The `PERSISTENCE` doesn't change in runtime, correct?
Otherwise, we might want to change this in the follow-up to avoid the instantiation.

Reply:
Also, by utilizing the injected configuration instance, it won't update at runtime.
It would need the same approach as in the JGroups instance and retrieve the configuration on each operation.

Reply:
According to the docs PERSISTENCE is runtime enablable, I put my comments as to how this could change. This just shows that this isn't being tested and I am not 100% certain that enabling tracing is going to work for all the cases.

Reply:
@jabolina good catch! the property persistence is supposed to be changed at runtime. Only persistence (since it is a global attribute cannot be changed at runtime).

going to d that:
> It would need the same approach as in the JGroups instance and retrieve the configuration on each operation.

Thank you!

Reply:
we need to add a test for it. thank you

Reply:
I added a test here: https://github.com/infinispan/infinispan/pull/11308/commits/15c8ee3c8509f7f2f6e72e37626bae9943586d99